### PR TITLE
Update Oracle connector documentation

### PR DESCRIPTION
**Pull request by rosewms**
Update Oracle connector documentation

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 1]
```diff
@@ -23,47 +22,363 @@ connection properties as appropriate for your setup:
...
+--------
+
+The connector supports :doc:`pushdown </optimizer/pushdown>` for optimized query
+processing.
 
 Limitations
 -----------
 
-The following SQL statements are not yet supported:
+Presto supports connecting to Oracle Database 12c. The following SQL statements
```
Comment (tooptoop4) [inline presto-docs/src/main/sphinx/connector/oracle.rst:376]: 12c is not the only version, 11g works, maybe others too

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 2]
```diff
@@ -23,47 +22,363 @@ connection properties as appropriate for your setup:
...
 
-If you used a different name for your catalog properties file, use
-that catalog name instead of ``oracle`` in the above examples.
+Mapping data types between Presto and Oracle
+--------------------------------------------
+
+Oracle and Presto support types that the other doesn't, the following tables
+show their type mapping:
+
+.. list-table:: Type mapping properties
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:95]: Maybe create a specific section for this table:
```
Type mapping configuration properties
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Also it feels to me that it would make sense to make it the last section.
After `Oracle to Presto type mapping` and `Presto to Oracle type mapping`

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 3]
```diff
@@ -23,47 +22,363 @@ connection properties as appropriate for your setup:
...
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p)``
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p) WITH TIME ZONE``
+    - ``TIMESTAMP WITH TIME ZONE``
+    - See :ref:`datetime mapping`
+
+If an Oracle table contains a type that's not listed in the above table, an
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:211]: This one is rendered weirdly with extra indentatinon. Not sure how to fix that.

image

Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:211]: This section requires major rephrasing

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 4]
```diff
@@ -23,47 +22,363 @@ connection properties as appropriate for your setup:
...
+   (default configuration), a column value converts to an unbounded ``VARCHAR``.
+
+Presto to Oracle type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto supports creating tables with the following types in an Oracle database.
+The table shows the mappings from Presto to Oracle data types:
+
+.. note::
+   Oracle connectors can't write Presto types that aren't listed in the table.
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:228]: I think it should say that for types not listed below Presto cannot perform `CREATE TABLE ... AS SELECT ..` operation. Inserting into alerady existing table uses `Oracle to Presto` type mapping.

cc: @findepi are there any quirks here?

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 5]
```diff
@@ -23,47 +22,363 @@ connection properties as appropriate for your setup:
...
+
+ - No precision is specified for the column (example: ``NUMBER`` or
+   ``NUMBER(*)``), unless ``oracle.number.default-scale`` is set. 
+ - Scale (``s`` ) is greater than precision. 
+ - Precision (``p`` ) is greater than 38. 
+ - Scale is negative and the difference between ``p`` and ``s`` is greater than
+   38, unless ``oracle.number.rounding-mode`` is set to a different value than
+   ``UNNECESSARY``. 
+
+If a column meets any of these conditions, it shows an error:
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:296]: again. I have hard time understanding this one.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 6]
```diff
@@ -23,47 +22,363 @@ connection properties as appropriate for your setup:
...
+For Oracle ``NUMBER`` (without precision and scale), you can change
+``oracle.number.default-scale=s`` and map the column to ``DECIMAL(38, s)``.
+
+.. _datetime mapping:
+
+Mapping datetime types
+^^^^^^^^^^^^^^^^^^^^^^
+
+Selecting a timestamp with fractional second precision (``p``) greater than 3
+truncates the fractional seconds to three digits instead of rounding it.
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:316]: I would drop `instead of rounding it`.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 7]
```diff
@@ -23,47 +22,363 @@ connection properties as appropriate for your setup:
...
+
+Oracle ``DATE`` values may store hours, minutes, and seconds, mapped
+to Presto ``TIMESTAMP``.
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:319]:
```suggestion
Oracle ``DATE`` type may store hours, minutes, and seconds, so it is mapped
to Presto ``TIMESTAMP``.
```

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 3 above)
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:211]: I'm not sure how to fix this but I'm working on it. It's structured the same way the other lists in the document are structured.

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 1 above)
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:376]: Removed reference in new draft.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 8]
```diff
@@ -2,19 +2,18 @@
...
-To configure the Oracle connector, create a catalog properties file
-in ``etc/catalog`` named, for example, ``oracle.properties``, to
-mount the Oracle connector as the ``oracle`` catalog.
-Create the file with the following contents, replacing the
-connection properties as appropriate for your setup:
+To configure the Oracle connector, create a catalog properties file in
+``etc/catalog`` named ``oracle.properties``. 
+
+To mount the Oracle connector as the ``oracle`` catalog, create a file with the
+following connection properties:
```
[the remark this answers is not in the document]
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:16]: I drafted an edited version of your suggestion.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 9]
```diff
@@ -23,47 +22,363 @@ connection properties as appropriate for your setup:
...
+To add another Oracle catalog, create a new properties file. For example, if
+you name the property file ``sales.properties``, Presto creates a catalog named
+sales.
 
 Querying Oracle
 ---------------
 
-The Oracle connector provides a schema for every Oracle *database*.
-You can see the available Oracle databases by running ``SHOW SCHEMAS``::
+The Oracle connector provides a schema for every Oracle server. Use the below
```
[the remark this answers is not in the document]
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:58]: Thanks for the explanation, it was helpful.

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 4 above)
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:228]: I edited based on your note, lemme know what you think of the new version.

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 3 above)
Comment (losipiuk) [reply presto-docs/src/main/sphinx/connector/oracle.rst:158]: The problem that it is hard to follow (at least to me) that it says about different possible outcomes, depending on value of configuration property `unsupported-type.handling-strategy`.
Also I have just noticed that each option is marked with "(default configuration)" :).
Also it should be `unsupported-type.handling` (without `-strategy`); The version with `-strategy` would work too, but it is legacy and will be removed eventually.

I would expect something like:

```
You can specify Presto behaviour for situation when type not listed above is used in Oracle table by setting `unsupported-type.handling-strategy` configuration property.

If ``unsupported-type.handling`` is set to ``FAIL``  querying table with unsupported table will fail.

If ``unsupported-type.handling`` is set to ``IGNORE`` (default configuration) columns with unsupported types will not be exposed in Presto.

If ``unsupported-type.handling`` is set to ``CONVERT_TO_VARCHAR`` column will be exposed as unbounded VARCHAR
```

Of course please make it more English :)

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 10]
```diff
@@ -23,47 +20,362 @@ connection properties as appropriate for your setup:
...
+Presto to Oracle type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto supports creating tables with the following types in an Oracle database.
+The table shows the mappings from Presto to Oracle data types:
+
+.. note::
+   For types not listed in the table below, Presto can't perform the ``CREATE
+   TABLE`` and ``AS SELECT``operations to insert info into existing
+   tables that use ``Oracle to Presto`` type mapping.
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:177]: this is single operation `CREATE TABLE <some_table> AS SELECT ....`

```suggestion
   For types not listed in the table below, Presto can't perform the ``CREATE
   TABLE <table> AS SELECT`` operations. 
   When data is inserted into existing tables ``Oracle to Presto`` type mapping is used.
```

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 11]
```diff
@@ -23,47 +20,362 @@ connection properties as appropriate for your setup:
...
+   ``UNNECESSARY``. 
+
+If a column meets any of the conditions:
+
+ - An ``unsupported-type.handling-strategy`` equals ``FAIL``
+   (default configuration) any queries on the table fail.
+ - An ``unsupported-type.handling-strategy`` equals ``IGNORE``
+   (default configuration), Presto no longer accesses the column.
+ - An ``unsupported-type.handling-strategy`` equals ``CONVERT_TO_VARCHAR``
+   (default configuration), a column value converts to an unbounded ``VARCHAR``.
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:252]: Please remove this one.
It is not related for `Presto to Oracle Type Mapping`. I think someone copy pasted this initially incidentally.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 12]
```diff
@@ -23,47 +20,362 @@ connection properties as appropriate for your setup:
...
+        positive infinity.
+      - ``FLOOR`` - Rounding mode to round towards negative
+        infinity.
+      - ``HALF_DOWN`` - Rounding mode to round towards
+        ``nearest neighbor`` unless both neighbors are
+        equidistant, in which case rounding down is used.
+      - ``HALF_EVEN`` - Rounding mode to round towards the
+        ``nearest neighbor`` unless both neighbors are
+        in which case rounding towards the nearest neighbor is
+        performed.
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:346]: I made suggestion fix here before. The rounding here is done differently.
```suggestion
      - ``HALF_EVEN`` - Rounding mode to round towards the
        ``nearest neighbor`` unless both neighbors are
        equidistant, in which case rounding towards the even neighbor is
        performed.
```

**Review by losipiuk — COMMENTED** [review]
Thanks. This is major improvement over what we had.
I mage a few suggestions, but overly LGTM.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 13]
```diff
@@ -2,19 +2,16 @@
 Oracle Connector
 ================
 
-The Oracle connector allows querying and creating tables in an
-external Oracle database. This can be used to join data between
-different systems like Oracle and Hive, or between two different
-Oracle instances.
+The Oracle connector allows querying and creating tables in an external Oracle
+database. Connectors let Presto join data provided by different databases,
```
[the remark this answers is not in the document]
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:6]: Most of the intros for the connectors start with a similar opening, it's mostly to orient people to where they are in the docs.

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 12 above)
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:346]: ooh, I actually did fix that but looking back at the doc, there is more than one instance where similar language is used.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 14]
```diff
@@ -23,47 +20,362 @@ connection properties as appropriate for your setup:
...
+    oracle.connection-pool.max-size=30
+    oracle.connection-pool.min-size=1
+    oracle.connection-pool.inactive-timeout=20m
+
+By default, the Oracle connector uses connection pooling for performance
+improvement. To disable connection pooling, update properties to include:
+
+.. code-block:: none
+
+    oracle.connection-pool.enabled=false
```
[the remark this answers is not in the document]
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:37]: Added to edits.

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 10 above)
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:177]: Added to edits.

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 11 above)
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:252]: Removed.

**Comment by losipiuk** [comment]
Please address https://github.com/prestosql/presto/pull/5014#discussion_r481428973. It is marked as resolved but no changes are applied.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 15]
```diff
@@ -23,47 +20,353 @@ connection properties as appropriate for your setup:
...
+  * - ``TIMESTAMP WITH TIME ZONE``
+    - ``TIMESTAMP(3) WITH TIME ZONE``
+    - See :ref:`datetime mapping`
+
+.. _number mapping:
+
+Mapping numeric types
+^^^^^^^^^^^^^^^^^^^^^
+
+An Oracle ``NUMBER(p, s)`` maps to Presto's ``DECIMAL(p, s)`` except in these
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:234]: This section renders incorrectly:

image

Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:235]: Fixed

**Comment by rosewms** [comment]
@losipiuk I forgot to hit submit on this comment before I pushed the updates it relates to...

Okay, I removed `-strategy` for future proofing, is there a reason to keep it if its going to be removed?

I also edited the Intro sentence to mention that examples are default configurations.

Let me know if the intro and the bullets are more understandable now?

**Review by mosabua — APPROVED** [review]
This add a lot of relevant information. Nice work. I hope we can get that merged for 341.

**Comment by electrum** [comment]
Let's replace the commit message with a single

> Update Oracle connector documentation

It's better to mention what rather than a specific file that was changed. The rest of the commit message is not very useful as it's a synopsis of the diff itself.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 16]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
+--------
+
+The connector supports :doc:`pushdown </optimizer/pushdown>` for optimized query
+processing.
 
 Limitations
 -----------
 
-The following SQL statements are not yet supported:
+The following SQL statements aren't supported:
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:366]: Using a contraction here looks weird, since this is a formal statement. Contractions are fine in prose, but for a short single statement like this, I don't see a benefit to using it.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 17]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
+
+.. code-block:: none
+
+    oracle.synonyms.enabled=true
+
+Pushdown
+--------
+
+The connector supports :doc:`pushdown </optimizer/pushdown>` for optimized query
+processing.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:361]: Don't wrap a single word as it looks funny and makes it harder to read. There is no strict line length limit.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 18]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
+      - ``UP`` - Rounding mode to round towards zero.
+      - ``DOWN`` - Rounding mode to round towards zero.
+
+    - ``UNNECESSARY``
+
+Synonyms
+--------
+
+Based on performance reasons, Presto disables support for Oracle ``SYNONYM``. To
+include ``SYNONYM``, add the following:
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:351]: "add the following configuration property"

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 19]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
+To access the ``clicks`` table in the ``web`` database run::
 
     SELECT * FROM oracle.web.clicks;
 
-If you used a different name for your catalog properties file, use
-that catalog name instead of ``oracle`` in the above examples.
+Mapping data types between Presto and Oracle
+--------------------------------------------
+
+Oracle supports types that Presto doesn't and Presto supports types that Oracle
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:89]: How about

> Both Oracle and Presto have types that are not supported by the Oracle connector.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 20]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
-You can see a list of the columns in the ``clicks`` table in the ``web`` database
-using either of the following::
+To see a list of the columns in the ``clicks`` table in the ``web``
+database run either of the following::
 
     DESCRIBE oracle.web.clicks;
     SHOW COLUMNS FROM oracle.web.clicks;
 
-Finally, you can access the ``clicks`` table in the ``web`` database::
+To access the ``clicks`` table in the ``web`` database run::
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:82]: The original structure seems better. It looks strange to have "run" at the end of the sentence, since it's missing a comma and seems to be missing something. The correct way to write a sentence like this would be

> To access the clicks table in the web database, run the following:

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 21]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
-in this database by running ``SHOW TABLES``::
+If you have an Oracle database named ``web``, run ``SHOW TABLES`` to see the
+tables it contains::
 
     SHOW TABLES FROM oracle.web;
 
-You can see a list of the columns in the ``clicks`` table in the ``web`` database
-using either of the following::
+To see a list of the columns in the ``clicks`` table in the ``web``
+database run either of the following::
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:77]: Missing comma

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 22]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
-You can see the available Oracle databases by running ``SHOW SCHEMAS``::
+The Oracle connector provides a schema for every Oracle database. Use the below
+queries to test your setup.
+
+If you used a different name for your catalog properties file, use that catalog
+name instead of ``oracle``.
+
+.. note::
+    The user configuration, in the connection properties file, determines your
+    privileges in these schemas. If the Oracle user doesn't have access to a
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:64]: The sentence seems clearer if we reverse it

> The Oracle user must have access to the table in order to access it from Presto.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 23]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
+improvement. The below configuration shows the typical default values. To update
+them, change the properties in the catalog configuration file:
+
+.. code-block:: properties
+
+    oracle.connection-pool.max-size=30
+    oracle.connection-pool.min-size=1
+    oracle.connection-pool.inactive-timeout=20m
+
+To disable connection pooling, update properties to include:
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:33]: "to include the following"

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 24]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
+
+.. code-block:: none
+
+    oracle.connection-pool.enabled=false
+
+.. note::
+
+    Include catalog files on all of your nodes, this requires you to restart.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:41]: This is two sentences separated with a comma

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 25]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
+To add another Oracle catalog, create a new properties file. For example, if
+you name the property file ``sales.properties``, Presto creates a catalog named
+sales.
 
 Querying Oracle
 ---------------
 
-The Oracle connector provides a schema for every Oracle *database*.
-You can see the available Oracle databases by running ``SHOW SCHEMAS``::
+The Oracle connector provides a schema for every Oracle database. Use the below
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/oracle.rst:56]: This last sentence is misleading, since it's not clear what "below queries" refers to. `SHOW SCHEMAS` will always work. The rest are examples, not something you could directly use.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 26]
```diff
@@ -23,47 +20,354 @@ connection properties as appropriate for your setup:
...
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p)``
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p) WITH TIME ZONE``
+    - ``TIMESTAMP WITH TIME ZONE``
+    - See :ref:`datetime mapping`
+
+Use the default configureation property, ``unsupported-type.handling``, to specify
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/connector/oracle.rst:158]:
```suggestion
Use the ``unsupported-type.handling`` configuration property, to specify
```

**Comment by losipiuk** [comment]
> @losipiuk I forgot to hit submit on this comment before I pushed the updates it relates to...
> 
> Okay, I removed `-strategy` for future proofing, is there a reason to keep it if its going to be removed?
> 
> I also edited the Intro sentence to mention that examples are default configurations.
> 
> Let me know if the intro and the bullets are more understandable now?

Thanks. It looks ok now. There was one typo (see the comment).
LGTM

**Review by losipiuk — APPROVED** [review]
LGTM module some styling suggestion already voiced.

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 16 above)
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:366]: I was using contractions based on the Google Style Guide: https://developers.google.com/style/contractions

> Negation contractions
In particular, it's fine to use -n't contractions, such as isn't, don't, and can't.

One reason that such contractions are useful is that it's sometimes easy for a reader to miss the word **not**, whereas it's harder to misread don't as do.

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 25 above)
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:56]: I restructured this section, hopefully its more clear.

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 26 above)
Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:158]: added the edit to my bulk edits.

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 27]
```diff
@@ -23,47 +20,355 @@ connection properties as appropriate for your setup:
...
+
+Presto to Oracle type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto supports creating tables with the following types in an Oracle database.
+The table shows the mappings from Presto to Oracle data types:
+
+.. note::
+   For types not listed in the table below, Presto can't perform the ``CREATE
+   TABLE <table> AS SELECT`` operations. When data is inserted into existing
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/connector/oracle.rst:179]: This sentence was suggested by @losipiuk .. however I am not sure this actually true. At least all other connector docs suggest that the Presto to Oracle mapping is used for ALL write operations .. and not just for CTAS ... 

Can you clarify if the Oracle connector is just different or if all other docs are wrong or whatever is technically correct @losipiuk @findepi and @electrum

**Review by mosabua — APPROVED** [review]
One last clarification needed, once fixed this is good to go imho

File: presto-docs/src/main/sphinx/connector/oracle.rst (hunk 27 above)
Comment (losipiuk) [reply presto-docs/src/main/sphinx/connector/oracle.rst:179]: The "Presto to <some DB>" mapping is used only in CTAS flow, when the external table does not yet exist, and we need to build the a schema for it based on Presto types of data used in CTAS statement.
For query operations as well as INSERT operations the "<some DB> to Presto" mapping is used. We already have columns in external database and we map types of those columns to Presto types.
This is not Oracle specific, but true for all connectors based on presto-base-jdbc.

Still I am not at all defending the wording I suggested. It is highly probable that one can phrase it much better :)

Hera are the code pointers:
* CTAS: https://github.com/prestosql/presto/blob/fd66bb72ddb9cfc311511900b1454cef385f172d/presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcPageSink.java#L74
* INSERT: https://github.com/prestosql/presto/blob/fd66bb72ddb9cfc311511900b1454cef385f172d/presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcPageSink.java#L89

Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/oracle.rst:179]: So looking at the code I conclude that this is true for all connectors that use the base-jdbc stuff .. so including mysql, postgresql, oracle, sqlserver and maybe others. 

At this stage I suggest we leaven the wording as it is in this PR and create another PR that clarifies this and corrects the docs for all affected connectors. Do you agree with this approach @losipiuk  and @electrum  .. if yes .. can we get this PR merged.

Comment (losipiuk) [reply presto-docs/src/main/sphinx/connector/oracle.rst:179]: SGTM

**Review by electrum — APPROVED** [review]

File: presto-docs/src/main/sphinx/connector/oracle.rst [hunk 28]
```diff
@@ -23,47 +20,350 @@ connection properties as appropriate for your setup:
...
 
 Limitations
 -----------
 
-The following SQL statements are not yet supported:
+The following SQL statements are not supported:
 
 * :doc:`/sql/delete`
+* :doc:`/sql/alter-table`
+* :doc:`/sql/create-table` (:doc:`/sql/create-table-as` is supported)
```
Comment (rosewms) [inline presto-docs/src/main/sphinx/connector/oracle.rst:366]: @losipiuk can you confirm that `Create table` isn't supported and that `create tables as` is supported? David asked me to double check. Thanks!

Comment (losipiuk) [reply presto-docs/src/main/sphinx/connector/oracle.rst:366]: Just checked. Create table is also supported.
@findepi do we have a test coverage for that? Quickly I could not find any.

Something as simple as
```java
    @Test
    public void testCreateTable()
    {
        getQueryRunner().execute("CREATE TABLE t (c BIGINT)");
        getQueryRunner().execute("INSERT INTO t VALUES(10)");
        assertThat(query("SELECT CAST(c as VARCHAR) from t")).matches("VALUES (VARCHAR '10')");
    }
```

or 

```java
    @Test
    public void testCreateTable()
    {
        getQueryRunner().execute("CREATE TABLE t (c BIGINT)");
        assertThat(query("DESCRIBE t")).matches(results -> results.getRowCount() > 0);
    }
```

Comment (findepi) [reply presto-docs/src/main/sphinx/connector/oracle.rst:366]: `io.prestosql.testing.AbstractTestDistributedQueries#testCreateTable`

Comment (rosewms) [reply presto-docs/src/main/sphinx/connector/oracle.rst:366]: @losipiuk thanks so much! I'm going to create a PR to remove `create tables` and the note about `create tables as` from the limitations section.

Comment (losipiuk) [reply presto-docs/src/main/sphinx/connector/oracle.rst:366]: > io.prestosql.testing.AbstractTestDistributedQueries#testCreateTable

Cool. I was looking in the `*Smoke*` ones.
